packages/components/src/components/hds/advanced-table/helpers.ts

```diff
@@ -0,0 +1,246 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import type { HdsAdvancedTableTdSignature } from './td';
+import type { HdsAdvancedTableThSignature } from './th';
+
+const getAllFocusableChildren = (element: HTMLElement): NodeListOf<Element> => {
+  return element.querySelectorAll(
+    'button:not([disabled]), a[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), [tabindex]:not([tabindex="-1"]):not([disabled])'
+  );
+};
+
+const getParentCell = (element: HTMLElement): Element | null | undefined => {
+  const targetParent = element.parentElement;
+
+  if (targetParent) {
+    return targetParent.closest(
+      '.hds-advanced-table__th, .hds-advanced-table__td'
+    );
+  }
+};
+
+const handleGridCellChildKeyPress = (event: KeyboardEvent): void => {
+  const { key, target } = event;
+
+  if (target instanceof HTMLElement) {
+    const cell = getParentCell(target);
+    if (cell instanceof HTMLElement) {
+      const allCellFocusableChildren = Array.from(
+        getAllFocusableChildren(cell)
+      ) as HTMLElement[];
```

Comment:
question: would this list become obsolete/outdated if elements are added to/removed from this cell?